src/H5private.h

```diff
@@ -49,29 +34,28 @@
 #include <limits.h>
 #include <math.h>
 #include <signal.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-/*
- * If _POSIX_VERSION is defined in unistd.h then this system is Posix.1
- * compliant. Otherwise all bets are off.
- */
+/* POSIX headers */
```

Comment:
The presence of unistd.h is fine for identifying POSIX-ness and I plan to propagate this idea throughout the library. We only need ifdef hacks where there's known problems and not for common things like fork().